runtime/development/src/lib.rs

```diff
@@ -2187,15 +2155,15 @@ impl_runtime_apis! {
 	}
 
 	// RewardsApi
-	impl runtime_common::apis::RewardsApi<Block, AccountId, Balance, RewardDomain, CurrencyId> for Runtime {
-		fn list_currencies(account_id: AccountId) -> Vec<(RewardDomain, CurrencyId)> {
+	impl runtime_common::apis::RewardsApi<Block, AccountId, Balance, CurrencyId> for Runtime {
+		fn list_currencies(account_id: AccountId) -> Vec<CurrencyId> {
 			pallet_rewards::Pallet::<Runtime, pallet_rewards::Instance1>::list_currencies(&account_id)
 			.into_iter().chain(
 				pallet_rewards::Pallet::<Runtime, pallet_rewards::Instance2>::list_currencies(&account_id).into_iter()
 			).collect()
 		}
 
-		fn compute_reward(currency_id: (RewardDomain, CurrencyId), account_id: AccountId) -> Option<Balance> {
+		fn compute_reward(currency_id: CurrencyId, account_id: AccountId) -> Option<Balance> {
 			<pallet_rewards::Pallet::<Runtime, pallet_rewards::Instance1> as AccountRewards<AccountId>>::compute_reward(currency_id, &account_id)
 			.or_else(|_|
 				<pallet_rewards::Pallet::<Runtime, pallet_rewards::Instance2> as AccountRewards<AccountId>>::compute_reward(currency_id, &account_id)
```

Comment:
I think this can no longer be chained, because we do not know from which pallet is comming the Currency. 

I think here we need to have two RPC calls like `collator_list_currencies` and `liquidity_list_currencies` or even better returning something like HashMap<String, Vec<CurrencyId>>`, with some identificatory name. Or using the Domain concept just as the key here.

Reply:
Great catch! I went with your third proposal of using the Domain concept as a key. The main reason for this is that using a HashMap introduced custom `encoding` and `decoding` complexity as it not available by default. 

[2a25440](https://github.com/centrifuge/centrifuge-chain/pull/1344/commits/2a254407c6165621f02a456b9818007192cd0007)